hs_core/management/commands/modify_resource_id.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+
+"""
+Modify the resource id of an existing resource
+
+"""
+
+from django.core.management.base import BaseCommand, CommandError
+from django.core.exceptions import ObjectDoesNotExist
+from hs_core.models import BaseResource, short_id
+from uuid import UUID
+from django.db import transaction, IntegrityError
+
+
+class Command(BaseCommand):
+    help = "Modify the resource id of an existing resource"
+
+    def add_arguments(self, parser):
+
+        # a list of resource id's, or none to check all resources
+        parser.add_argument('resource_id', type=str, help=('Required. The existing id (short_id) of'
+                                                           ' the resource'))
+        parser.add_argument('new_resource_id', type=str,
+                            help=('Optional. The new id (short_id) for the resource. A random one '
+                                  'is generated if none is provided. Must be a valid string '
+                                  'representation of a uuid hex'))
+
+    def handle(self, *args, **options):
+
+        if not options['resource_id']:
+            raise CommandError('resource_id argument is required')
+        res_id = options['resource_id']
+        try:
+            res = BaseResource.objects.get(short_id=res_id)
+        except ObjectDoesNotExist:
+            raise CommandError("No Resource found for id {}".format(res_id))
+
+        if options['new_resource_id']:
+            try:
+                UUID(options['new_resource_id'])
+                new_res_id = options['new_resource_id']
+            except Exception as e:
+                raise CommandError('new_resource_id {} must be a valid uuid hex string'
+                                   .format(options['new_resource_id']), e)
+
+            try:
+                if BaseResource.objects.get(short_id=new_res_id):
+                    raise CommandError('resource with id {} already exists'.format(new_res_id))
+            except ObjectDoesNotExist:
+                pass
+        else:
+            new_res_id = short_id()
+
+        try:
+            with transaction.atomic():
+                print("Deleting existing bag")
+                res.bags.all().delete()
+                res.setAVU("bag_modified", True)
```

Comment:
@sblack-usu I believe setting bag_modified to True would allow the new resource bag to be regenerated, but the old short_id.zip bag file would still be there hanging around since res.bags.all().delete() only deletes bag metadata in Django without deleting the zip file in iRODS. I'd suggest to move the old zip file if it exists, or add an additional step to delete the old short_id.zip if it exists.

Reply:
you're right, I should have checked that method to make sure it actually deletes the bag.  It doesn't make sense to move the bag because old id is all over the bag in folders and metadata.  Easier to just delete it.

Reply:
Agreed - forgot the old bag actually includes the old id so cannot be carried over to represent the new bag.

Reply:
The latest commit addressed the old bag as well as sets the metadata_dirty flag.  It also occurred to me that aggregations have their own way of writing metadata, so those will have to be regenerated as well.  @pkdash - what's the best way to handle aggregation metadata?

Reply:
I think setting metadata_dirty flag to get metadata xml file regenerated from Django DB would most likely suffice, but @pkdash can confirm whether this indeed suffice or not for aggregations.

Reply:
I updated the branch to recreate the aggregation xml files.  @pkdash - could you take a look at this and let me know if aggregations are taken care of completely?  I ran a test and am pretty sure all references to the resource id are being updated.

Reply:
Looks good to me.